### PR TITLE
Add test for glDrawElements with data on client side

### PR DESCRIPTION
**Pull request by caiiiycuk**
Hi guys! I found strange issue with LEGACY_GL_EMULATION. I wrote the test case to demonstrate it. This code should render triangle on yellow background and actually works in Chrome, Firefox, but not in Safari < 10. I've tested on Safari 8, 9 and 10 (works). In old Safari this code renders only yellow background (console does not have any error). Moreover if you replace 
```glDrawElements(GL_TRIANGLES, 3, GL_UNSIGNED_SHORT, indices);```
with
```glDrawArrays(GL_TRIANGLES, 0, 3);```
then Safari also will render correct. So I think the problem is near prepareClientAttributes, but how figure it out I don' know. Need help?

P. S.  I think that test is good and it will be good to commit it. Thx.

### CODE REVIEW

**Comment by caiiiycuk** [comment]
I've checked that if I replace client side arrays with vertex buffer object and index buffer object that Safari 8,9 works fine. I think something went wrong in LEGACY_GL_EMULATION code

File: tests/gl_draw_elements.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    glLoadIdentity();
+
+    glMatrixMode(GL_MODELVIEW);
+    glLoadIdentity();
+
+    glClearColor(1, 1, 0, 1);
+    glClear(GL_COLOR_BUFFER_BIT);
+
+    glEnableClientState(GL_VERTEX_ARRAY);
+    glEnableClientState(GL_COLOR_ARRAY);
```
Comment (juj) [inline tests/gl_draw_elements.c]: Old fixed pipeline GL has already begun to escape my brain, I wonder if this needs a `glEnableClientState(GL_TEXTURE_COORD_ARRAY)` as well?

File: tests/gl_draw_elements.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+typedef struct Vertex {
+    short           x, y, z;
+    unsigned char   c_r, c_g, c_b, c_a;
+    short           u,v;
+} Vertex;
+
+Vertex triangle[3] = {
+    0, 0, 0,   0, 255, 255, 255, 0, 0,
+    1, 1, 0, 255,   0, 255, 255, 0, 0,
+    1, 0, 0, 255, 255,   0, 255, 0, 0
```
Comment (juj) [inline tests/gl_draw_elements.c]: Above, struct Vertex defines x, y, z, u and v as `short`s, but it looks like this array fills them in as if they were `unsigned char`s. Is that intentional for this test to provide this kind of a mismatched input array?

Comment (juj) [inline tests/gl_draw_elements.c]: Err, nevermind the above, I was somehow reading this to be filling raw bytes, but it's filling elements, so all good.

**Comment by juj** [comment]
If new Firefox, Chrome and Safari all work correctly, but old Safari does not, wouldn't this be a Safari bug then? Is there something specific that suggests that Emscripten's fixed GL emulation would have a bug?

File: tests/gl_draw_elements.c (hunk 1 above)
Comment (caiiiycuk) [inline tests/gl_draw_elements.c]: You are right, but for this test texturing is not needed, so I remove u,v coordinates from Vertex structure.

**Comment by caiiiycuk** [comment]
> Is there something specific that suggests that Emscripten's fixed GL emulation would have a bug?

Not sure about it. Originally problem occurs in big game that I port in js. When I use old way to render graphics (like in the test), then graphics not rendered at all in Safari 8,9. When I rewrite most of code to rendered with vbo+index buffer  then this code start render correctly, but some other places stop to drawing even in chrome and ff until I rewrite all code to vbo+index buffer. 

As I understand the legacy emulation for this simple test should do the same thing that I do for render data in Safari 8,9. Create vbo + index buffer, fill this buffer from memory data and then call to glDrawElements. If it is correct then in Safari 8,9 this test should pass.

Btw, definitely there are problem in Safari and it is not critical for me (because I already rewrite the code), but may be there are some hidden problem in emscripten emultaion layer. However it will be good to merge this test.

**Pull request closed without merging**